src/System.Net.Security/tests/FunctionalTests/NegotiateStreamTestForUnix.cs

```diff
@@ -453,6 +475,295 @@ public void NegotiateStream_StreamToStream_KerberosAuthInvalidTarget_Failure()
+                };
+        }
+
+        public static IEnumerable<object[]> InvalidNtlmCredentials
+        {
+            get
+            {
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword, TestConfiguration.Domain.Substring(1))};
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser.Substring(1), _ntlmPassword, TestConfiguration.Domain) };
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword.Substring(1), TestConfiguration.Domain) };
+                yield return new object[] { new NetworkCredential(string.Format(@"{0}\{1}", TestConfiguration.Domain, TestConfiguration.NtlmUser), _ntlmPassword, TestConfiguration.Domain.Substring(1))};
+                yield return new object[] { new NetworkCredential(string.Format("{0}@{1}", TestConfiguration.NtlmUser, TestConfiguration.Domain.Substring(1)), _ntlmPassword)};
+                yield return new object[] { new NetworkCredential(string.Format(@"{0}\{1}", TestConfiguration.Domain.Substring(1), TestConfiguration.NtlmUser), _ntlmPassword, TestConfiguration.Domain)};
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword, TestConfiguration.Realm)};
+
+            }
+        }
+
+
+        [Theory, OuterLoop]
+        [MemberData("ValidNtlmCredentials")]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success(NetworkCredential credential)
+        {
+            if (!_isNtlmAvailable)
+            {
+                _output.WriteLine("skipping NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success");
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new NegotiateStream(clientStream))
+            using (var server = new UnixGssFakeNegotiateStream(serverStream))
+            {
+                Assert.False(client.IsAuthenticated, "client is not authenticated");
+                Assert.False(server.IsAuthenticated, "server is not authenticated");
+
+                Task[] auth = new Task[2];
+                auth[0] = client.AuthenticateAsClientAsync(credential, _testTarget, ProtectionLevel.None, TokenImpersonationLevel.Identification);
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                bool finished = Task.WaitAll(auth, TestConfiguration.PassingTestTimeoutMilliseconds);
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated, "client is now authenticated");
```

Comment:
I thought we changed these before to be of the form

```C#
Assert.True(client.IsAuthenticated, "client.IsAuthenticated");
```

Since that still uniquely identifies the failure in the method, requires less poetical thinking, and as a side benefit makes Assert.False look right in an error message:

```
    Some.Example.Test.I.Just.Made [FAIL]
       client.IsEncrypted
       Expected: False
       Actual:   True
       Stack Trace: ...
```

But, since I see existing ones, I guess not.  It'd be great if you wanted to clean these up; but I can't say it's non-standard if you don't want to.